### PR TITLE
Add support for Airflow 2.0

### PR DESCRIPTION
**Pull request by kaxil**
## Description

> Describe the purpose of this pull request.

This PR adds support for running 2.0 and keeps backwards-compatibility will 1.10.x versions

## 🎟 Issue(s)

Resolves astronomer/issues#XXXX

## 🧪 Functional Testing

> List the functional testing steps to confirm this feature or fix.

## 📸 Screenshots

> Add screenshots to illustrate the validity of these changes.

## 📋 Checklist

- [x] Rebased from the main (or release if patching) branch (before testing)
- [ ] Ran `make test` before taking out of draft
- [ ] Added/updated applicable tests
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: airflow/docker_test.go [hunk 1]
```diff
@@ -112,7 +112,7 @@ services:
   webserver:
     image: test-project-name/airflow:latest
     command: >
-      bash -c "airflow create_user -r Admin -u admin -e admin@example.com -f admin -l user -p admin && airflow webserver"
+      bash -c 'airflow create_user "$@" || airflow users create "$@"' -- -r Admin -u admin -e admin@example.com -f admin -l user -p admin && airflow webserver"
```
Comment (kaxil) [inline airflow/docker_test.go:115]: Need some help here, do we need to escape `$` over here?

File: airflow/include/composeyml.go [hunk 2]
```diff
@@ -62,7 +62,7 @@ services:
   webserver:
     image: {{ .AirflowImage }}
     command: >
-      bash -c "airflow create_user -r Admin -u admin -e admin@example.com -f admin -l user -p admin && airflow webserver"
+      bash -c 'airflow create_user "$$@" || airflow users create "$$@"' -- -r Admin -u admin -e admin@example.com -f admin -l user -p admin && airflow webserver"
```
Comment (kaxil) [inline airflow/include/composeyml.go:65]: Need some help here, do we need to escape `$` over here? Without escaping the test was failing with interpolation error

File: airflow/docker_test.go (hunk 1 above)
Comment (andriisoldatenko) [reply airflow/docker_test.go:115]:
```
Env file ".env" found. Loading...
ERRO[0000] Could not parse config for project t20201013104455846271 : Invalid interpolation format for key "command": "bash -c 'airflow create_user "$@" || airflow users create "$@"' -- -r Admin -u admin -e admin@example.com -f admin -l user -p admin && airflow webserver"
"
Error: Error creating docker-compose project: Invalid interpolation format for key "command": "bash -c 'airflow create_user "$@" || airflow users create "$@"' -- -r Admin -u admin -e admin@example.com -f admin -l user -p admin && airflow webserver"
"
```
yeah it doesn't work... let me think a little bit.

Comment (andriisoldatenko) [reply airflow/docker_test.go:115]: @kaxil my version:
```
bash-5.0$ bash -c "args='-r Admin -u admin -e admin@example.com -f admin -l user -p admin';airflow create_user $args || airflow users create $args && airflow webserver"
[2020-10-13 08:21:13,767] {settings.py:254} INFO - settings.configure_orm(): Using pool settings. pool_size=5, max_overflow=10, pool_recycle=1800, pid=419

[2020-10-13 08:21:18,400] {__init__.py:51} INFO - Using executor LocalExecutor
[2020-10-13 08:21:18,401] {dagbag.py:403} INFO - Filling up the DagBag from /usr/local/airflow/dags
[2020-10-13 08:21:19,160] {security.py:461} INFO - Start syncing user roles.
[2020-10-13 08:21:19,454] {security.py:370} INFO - Fetching a set of all permission, view_menu from FAB meta-table
[2020-10-13 08:21:19,668] {security.py:313} INFO - Cleaning faulty perms
admin already exist in the db
[2020-10-13 08:21:21,013] {settings.py:254} INFO - settings.configure_orm(): Using pool settings. pool_size=5, max_overflow=10, pool_recycle=1800, pid=435
  ____________       _____________
 ____    |__( )_________  __/__  /________      __
____  /| |_  /__  ___/_  /_ __  /_  __ \_ | /| / /
___  ___ |  / _  /   _  __/ _  / / /_/ /_ |/ |/ /
 _/_/  |_/_/  /_/    /_/    /_/  \____/____/|__/
[2020-10-13 08:21:25,576] {__init__.py:51} INFO - Using executor LocalExecutor
[2020-10-13 08:21:25,577] {dagbag.py:403} INFO - Filling up the DagBag from /dev/null
^
```

Comment (ashb) [inline airflow/docker_test.go:115]:
```suggestion
      bash -c 'airflow create_user "$$@" || airflow users create "$$@"' -- -r Admin -u admin -e admin@example.com -f admin -l user -p admin && airflow webserver"
```

File: airflow/include/composeyml.go [hunk 3]
```diff
@@ -62,7 +62,7 @@ services:
   webserver:
     image: {{ .AirflowImage }}
     command: >
-      bash -c "airflow create_user -r Admin -u admin -e admin@example.com -f admin -l user -p admin && airflow webserver"
+      bash -c 'airflow create_user "$@" || airflow users create "$@"' -- -r Admin -u admin -e admin@example.com -f admin -l user -p admin && airflow webserver"
```
Comment (ashb) [inline airflow/include/composeyml.go:65]:
```suggestion
      bash -c 'airflow create_user "$$@" || airflow users create "$$@"' -- -r Admin -u admin -e admin@example.com -f admin -l user -p admin && airflow webserver"
```

File: airflow/docker_test.go [hunk 4]
```diff
@@ -112,7 +112,7 @@ services:
   webserver:
     image: test-project-name/airflow:latest
     command: >
-      bash -c "airflow create_user -r Admin -u admin -e admin@example.com -f admin -l user -p admin && airflow webserver"
+      bash -c 'airflow create_user "$$@" || airflow users create "$$@"' -- -r Admin -u admin -e admin@example.com -f admin -l user -p admin && airflow webserver"
```
Comment (ashb) [inline airflow/docker_test.go:115]:
```suggestion
      bash -c 'airflow create_user "$$@" || airflow users create "$$@"' -- -r Admin -u admin -e admin@example.com -f admin -l user -p admin && airflow webserver
```

File: airflow/include/composeyml.go (hunk 2 above)
Comment (ashb) [inline airflow/include/composeyml.go:65]:
```suggestion
      bash -c 'airflow create_user "$$@" || airflow users create "$$@"' -- -r Admin -u admin -e admin@example.com -f admin -l user -p admin && airflow webserver
```

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**